stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/factory/ChannelsViewModelFactory.kt

```diff
@@ -22,7 +22,10 @@ import io.getstream.chat.android.livedata.ChatDomain
 public class ChannelsViewModelFactory @JvmOverloads constructor(
     private val filter: FilterObject = Filters.and(
         Filters.eq("type", "messaging"),
-        Filters.`in`("members", listOf(ChatDomain.instance().currentUser.id)),
+        Filters.`in`(
+            "members",
+            ChatDomain.instance().user.value?.id?.let(::listOf) ?: emptyList<String>()
```

Comment:
I don't like this solution, but I'm not sure how to fix it. Default filter should always contain `currentUser.id` but instead, it's possible that we are going to use the wrong filter when querying channels.   @GetStream/android-developers what do you think?

Reply:
🤔  One possibility is to use `!!` so user should be always non null

Reply:
Buggy approach for ViewModels. But we don't have any alternative right now. Actually seems for offline support we shouldn't allow any screen initialization until we set value to `ChatDomain`. It's some complex problem that we can discuss on technical meeting